scripts/lua/cors.lua

```diff
@@ -25,24 +25,33 @@ local _M = {}
 local request = require "lib/request"
 
 function _M.processCall(resourceConfig)
-  if resourceConfig.cors ~= nil then
-    _M.setCorsHeaders(resourceConfig.cors.origin, resourceConfig.cors.methods)
-    if ngx.req.get_method() == "OPTIONS" then
+  ngx.var.cors_origins = resourceConfig.cors.origin
+  ngx.var.cors_methods = resourceConfig.cors.methods
+  if resourceConfig.cors ~= nil then 
+    if resourceConfig.cors.origin ~= 'false' then 
+     if ngx.req.get_method() == "OPTIONS" then
       request.success(200)
+     end
     end
   end
 end
 
-function _M.setCorsHeaders(corsOrigin, corsMethods)
-  if corsOrigin ~= nil then
-    if corsOrigin == 'false' then
+function _M.replaceHeaders()
+  if ngx.var.cors_origins ~= nil then
+    if ngx.var.cors_origins == 'true' then
+      ngx.header['Access-Control-Allow-Origin'] = '*'
+      if ngx.var.cors_methods ~= nil then
+        ngx.header['Access-Control-Allow-Methods'] = ngx.var.cors_methods
+      else
+        ngx.header['Access-Control-Allow-Methods'] = 'GET, POST, PUT, DELETE, PATCH, HEAD, OPTIONS'
```

Comment:
Instead of `else`ing this, why not just make it the default and override it if `cors_methods ~= nil`?